listener.go

```diff
@@ -54,113 +50,72 @@ func processSnapshot(snapshot *common.Snapshot) {
 
 }
 
-func processSqliteSnapshot(snapshot *common.Snapshot, db apid.DB) {
-	//nothing to do as of now, here as a placeholder
-}
-
-func processJsonSnapshot(snapshot *common.Snapshot, db apid.DB) {
-
-	err := initDB(db)
-	if err != nil {
-		log.Panicf("Unable to initialize database: %v", err)
-	}
+func processSqliteSnapshot(db apid.DB) {
 
-	tx, err := db.Begin()
+	var numApidClusters int
+	apidClusters, err := db.Query("SELECT COUNT(*) FROM edgex_apid_cluster")
 	if err != nil {
-		log.Panicf("Error starting transaction: %v", err)
+		log.Panicf("Unable to read database: %s", err.Error())
 	}
-	defer tx.Rollback()
-
-	for _, table := range snapshot.Tables {
-
-		switch table.Name {
-		case LISTENER_TABLE_APID_CLUSTER:
-			if len(table.Rows) > 1 {
-				log.Panic("Illegal state for apid_cluster. Must be a single row.")
-			}
-			for _, row := range table.Rows {
-				ac := makeApidClusterFromRow(row)
-				err := insertApidCluster(ac, tx)
-				if err != nil {
-					log.Panicf("Snapshot update failed: %v", err)
-				}
-			}
+	apidClusters.Next()
+	apidClusters.Scan(&numApidClusters)
 
-		case LISTENER_TABLE_DATA_SCOPE:
-			for _, row := range table.Rows {
-				ds := makeDataScopeFromRow(row)
-				err := insertDataScope(ds, tx)
-				if err != nil {
-					log.Panicf("Snapshot update failed: %v", err)
-				}
-			}
-		}
+	if numApidClusters != 1 {
+		log.Panic("Illegal state for apid_cluster. Must be a single row.")
 	}
 
-	err = tx.Commit()
+	_, err = db.Exec("ALTER TABLE edgex_apid_cluster ADD COLUMN last_sequence text DEFAULT ''")
 	if err != nil {
-		log.Panicf("Error committing Snapshot change: %v", err)
+		if err.Error() == "duplicate column name: last_sequence" {
+			return
+		} else {
+			log.Error("[[" + err.Error() + "]]")
+			log.Panicf("Unable to create last_sequence column on DB.  Unrecoverable error ", err)
+		}
 	}
 }
 
-func processChangeList(changes *common.ChangeList) {
+func processChangeList(changes *common.ChangeList) bool {
+
```

Comment:
Open a story (for future) to make this return ERRNO (enum) instead of bool. And to write cases to cover individual error case.